src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -122,20 +125,40 @@ public int Count
                     }
                     else if (index == 2)
                     {
-                        return new KeyValuePair<string, object>("ActivityId", _activityId);
+                        return new KeyValuePair<string, object>("SpanId", _spanId);
+                    }
+                    else if (index == 3)
+                    {
+                        return new KeyValuePair<string, object>("TraceId", _traceId);
+                    }
+                    else if (index == 4)
+                    {
+                        return new KeyValuePair<string, object>("ParentId", _parentId);
                     }
 
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
             }
 
-            public HostingLogScope(HttpContext httpContext, string activityId)
+            public HostingLogScope(HttpContext httpContext, Activity activity)
             {
                 _traceIdentifier = httpContext.TraceIdentifier;
                 _path = (httpContext.Request.PathBase.HasValue 
                          ? httpContext.Request.PathBase + httpContext.Request.Path 
                          : httpContext.Request.Path).ToString();
-                _activityId = activityId;
+
+                if (activity.IdFormat == ActivityIdFormat.W3C)
+                {
+                    _traceId = activity.TraceId.ToHexString();
+                    _spanId = activity.SpanId.ToHexString();
+                    _parentId = activity.ParentId;
```

Comment:
```suggestion
                    _parentId = activity.ParentSpanId;
```

Reply:
Allocations everywhere. Will this return null if ParentSpanId is null?

Reply:
nope :( it will be "0000000000000000". Need null check on ParentId before accessing it to avoid extra allocation. 

I guess we should fix it in `ActivitySpanId`. If `id1 == 0` return const "0000000000000000", similar for TraceId

https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1229